docs/src/main/asciidoc/ghpages.sh

```diff
@@ -12,6 +12,40 @@ if ! [ -d docs/target/generated-docs ]; then
     exit 0
 fi
 
+# The script should be executed from the root folder
+
+ROOT_FOLDER=`pwd`
+echo "Current folder is ${ROOT_FOLDER}"
+
+if [[ ! -e "${ROOT_FOLDER}/.git" ]]; then
+    echo "You're not in the root folder of the project!"
+    exit 1
+fi
+
+# Retrieve version number, name of the main adoc and name of the current branch
+###################################################################
+
+# Code grepping for the 1st presence of "version>" in pom.xml.
+# First one is project version, second parent version.
+VERSION_NODE=`awk '/version>/{i++}i==1{print; exit}' $ROOT_FOLDER/pom.xml`
+# Extract the contents of the version node
+VERSION_VALUE=$(sed -ne '/version/{s/.*<version>\(.*\)<\/version>.*/\1/p;q;}' <<< "$VERSION_NODE")
+echo "Extracted version from root pom.xml is [${VERSION_VALUE}]"
+
+# Code grepping for the 2nd presence of "version>" in pom.xml.
+# First one is parent, second project version.
```

Comment:
> docs can differ between minor versions can't they?

Yes, but we only support the latest version of the code, so I don't see any value in having the docs in a separate directory for unsupported versions. Tags will identify the old values if we need them for reference. So for sleuth at the moment we need 1.0.x and master (which doesn't need a sub directory).